crates/flight_client/src/lib.rs

```diff
@@ -629,7 +641,25 @@ impl FlightClient {
 
 #[allow(clippy::needless_pass_by_value)]
 fn map_tonic_error_to_message(e: tonic::Status) -> Error {
+    if is_connection_reset_error(&e) {
+        return Error::ConnectionReset {
+            source: TonicStatusError::from(e),
+        };
+    }
     Error::UnableToQuery {
         source: e.message().into(),
     }
 }
+
+pub fn is_connection_reset_error(error: &tonic::Status) -> bool {
+    if error.code() == tonic::Code::Internal {
+        let error_message = error.message().to_lowercase();
+        if error_message.contains("operation was canceled")
```

Comment:
is there a typo in the `tonic::Status` message or your `contains` invocation. "operation was cancelled"